pkg/server/router.go

```diff
@@ -42,14 +44,20 @@ func (r *Router) Routes(h router.Handlers) http.Handler {
 	loginHandler := auth.NewLoginHandler(r.scaled, r.restConfig)
 	m.Path("/v1-public/auth").Handler(loginHandler)
 	m.Path("/v1-public/auth-modes").HandlerFunc(auth.ModeHandler)
-	m.PathPrefix("/v3-public/").Handler(&proxy.Handler{})
-	m.PathPrefix("/v3/").Handler(&proxy.Handler{})
 
 	vueUI := ui.Vue
 	m.Handle("/dashboard/", vueUI.IndexFile())
 	m.PathPrefix("/dashboard/").Handler(vueUI.IndexFileOnNotFound())
 	m.PathPrefix("/api-ui").Handler(vueUI.ServeAsset())
 
+	if r.options.RancherEmbedded {
+		v3Handler := &proxy.Handler{
+			Host: r.options.RancherURL,
```

Comment:
Can we parse the URL and ensure the host is valid here? So that we don't need to panic in the handler when url is invalid